plan/physical_plan_builder.go

```diff
@@ -454,379 +456,410 @@ func (p *Join) convert2PhysicalPlan(prop requiredProperty) (*physicalPlanInfo, *
+			return nil, errors.Trace(err)
 		}
-		p.storePlanInfo(prop, sortedPlanInfo, unSortedPlanInfo, count)
-		return sortedPlanInfo, unSortedPlanInfo, count, nil
+		p.storePlanInfo(prop, info)
+		return info, nil
 	case RightOuterJoin:
-		sortedPlanInfo, unSortedPlanInfo, count, err := p.handleRightJoin(prop, false)
+		info, err := p.handleRightJoin(prop, false)
 		if err != nil {
-			return nil, nil, 0, errors.Trace(err)
+			return nil, errors.Trace(err)
 		}
-		p.storePlanInfo(prop, sortedPlanInfo, unSortedPlanInfo, count)
-		return sortedPlanInfo, unSortedPlanInfo, count, nil
+		p.storePlanInfo(prop, info)
+		return info, nil
 	default:
-		lSortedPlanInfo, lunSortedPlanInfo, count, err := p.handleLeftJoin(prop, true)
+		lInfo, err := p.handleLeftJoin(prop, true)
 		if err != nil {
-			return nil, nil, 0, errors.Trace(err)
+			return nil, errors.Trace(err)
 		}
-		rsortedPlanInfo, runSortedPlanInfo, _, err := p.handleRightJoin(prop, true)
+		rInfo, err := p.handleRightJoin(prop, true)
 		if err != nil {
-			return nil, nil, 0, errors.Trace(err)
+			return nil, errors.Trace(err)
 		}
-		if rsortedPlanInfo.cost < lSortedPlanInfo.cost {
-			lSortedPlanInfo = rsortedPlanInfo
+		if rInfo.cost < lInfo.cost {
+			lInfo = rInfo
 		}
-		if runSortedPlanInfo.cost < lunSortedPlanInfo.cost {
-			lunSortedPlanInfo = runSortedPlanInfo
+		p.storePlanInfo(prop, lInfo)
+		return lInfo, nil
+	}
+}
+
+func (p *Aggregation) handleStreamAgg(prop *requiredProperty) (*physicalPlanInfo, error) {
+	agg := &PhysicalAggregation{
+		AggType:      StreamedAgg,
+		AggFuncs:     p.AggFuncs,
+		GroupByItems: p.GroupByItems,
+	}
+	agg.HasGby = len(p.GroupByItems) > 0
+	agg.SetSchema(p.schema)
+	// TODO: Consider distinct key.
+	gbyCols := make([]*expression.Column, 0, len(p.GroupByItems)+1)
+	info := &physicalPlanInfo{cost: math.MaxFloat64}
+	// TODO: extract columns in monotonic functions.
+	for _, item := range p.GroupByItems {
+		if col, ok := item.(*expression.Column); ok {
+			if !col.Correlated {
+				gbyCols = append(gbyCols, col)
+			}
+		} else {
+			// group by a + b is not interested in any orders.
+			return info, nil
+		}
+	}
+	isSortKey := make([]bool, len(gbyCols))
+	newProp := &requiredProperty{
+		props: make([]*columnProp, 0, len(gbyCols)),
+	}
+	for _, pro := range prop.props {
+		isMatch := false
+		for i, col := range gbyCols {
+			if col.ColName.L == pro.col.ColName.L {
+				isSortKey[i] = true
+				isMatch = true
+			}
+		}
+		if !isMatch {
+			return info, nil
+		}
+		newProp.props = append(newProp.props, pro)
+	}
+	newProp.sortKeyLen = len(newProp.props)
+	for i, col := range gbyCols {
+		if !isSortKey[i] {
+			newProp.props = append(newProp.props, &columnProp{col: col})
+		}
+	}
+	childInfo, err := p.children[0].(LogicalPlan).convert2PhysicalPlan(newProp)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	info = addPlanToResponse(agg, childInfo)
+	info.cost += float64(info.count) * cpuFactor
+	info.count = info.count / 3
```

Comment:
Why the value is `3` here? And define a const is better. The same as the value in `estimateJoinCount`.